### ISSUE TITLE
verification error message lists unrelated precondition 

### ISSUE THREAD

**Issue by tjhance** [verus-lang/verus#114]
This:

```
#[allow(unused_imports)]
use builtin::*;
mod pervasive;
use pervasive::*;

fn get_bool() -> bool {
    ensures(|b: bool| b == true);
    true
}

fn require_false() -> bool {
    requires(false);                // line 12
    false
}

fn main() {
    let x = 6;

    if !get_bool() {
        require_false();
    }

    assert(x == 7);              // line 23
}
```

gives the following error message:

```
Verifying root module
error: assertion failure
  --> l.rs:23:5
   |
12 |     requires(false);
   |              ----- failed precondition
...
23 |     assert(x == 7);
   |     ^^^^^^^^^^^^^^
```

This doesn't make any sense at all - line 12 should easily pass verification, and also, it isn't even a precondition of the actual failing assert on line 23

---

* [ ] when fixed, un-ignore test added in https://github.com/secure-foundations/verus/commit/5c6c14d9dfa24c91905d79e708cd464be9ad478e

**Comment by tjhance** [comment]
(also just to emphasize: this isn't an erroneous proof failure - the error message is actually ill-formed)

**Comment by tjhance** [comment]
verus doesn't exhibit this behavior anymore. probably fine to close, though i don't know what fixed it, or if the problem is still lurking somewhere

**Comment by Chris-Hawblitzel** [comment]
I forgot to close the issue.  This was the commit that fixed it: 858001ec9cd9ff2e6fa4921cf2b7385f1c75dccc

**Comment by tjhance** [comment]
I've un-ignored the test (959ffdfecd2db0d8871d97a27598daf65b984731). Closing.
